.ci/watchdog/GitWrapper.py

```diff
@@ -0,0 +1,72 @@
+# The source code contained or described herein and all documents related to the
+# source code ("Material") are owned by Intel Corporation or its suppliers or
+# licensors. Title to the Material remains with Intel Corporation or its
+# suppliers and licensors. The Material may contain trade secrets and proprietary
+# and confidential information of Intel Corporation and its suppliers and
+# licensors, and is protected by worldwide copyright and trade secret laws and
+# treaty provisions. No part of the Material may be used, copied, reproduced,
+# modified, published, uploaded, posted, transmitted, distributed, or disclosed
+# in any way without Intel's prior express written permission.
+# No license under any patent, copyright, trade secret or other intellectual
+# property right is granted to or conferred upon you by disclosure or delivery of
+# the Materials, either expressly, by implication, inducement, estoppel or
+# otherwise. Any license under such intellectual property rights must be express
+# and approved by Intel in writing.
+# Include any supplier copyright notices as supplier requires Intel to use.
+# Include supplier trademarks or logos as supplier requires Intel to use,
+# preceded by an asterisk. An asterisked footnote can be added as follows:
+# *Third Party trademarks are the property of their respective owners.
+# Unless otherwise agreed by Intel in writing, you may not remove or alter
+# this notice or any other notice embedded in Materials by Intel or Intel's
+# suppliers or licensors in any way.
+
+from time import sleep
+from github import Github
+import datetime
+import logging
+
+# Logging
+log = logging.getLogger(__file__)
+ch = logging.StreamHandler()
+log.setLevel(logging.INFO)
+ch.setFormatter(logging.Formatter('%(name)s - %(levelname)s - %(message)s'))
+log.addHandler(ch)
+
+_RETRY_LIMIT = 3
+_RETRY_COOLDOWN = 15
+
+class GitWrapper:
+    def __init__(self, git_token, repository, project):
+        self.git = Github(git_token)
+        self.repository = repository
+        self.project = project
+
+    def _try_git(self, method, extraArgs=[]):
+        attempt = 0
+        while(attempt < _RETRY_LIMIT):
+            try:
+                result = method(*extraArgs)
+                return result
+            except:
+                attempt = attempt + 1
+                log.warning("Failed to execute " + method.__name__ + " attempt: " + str(attempt))
+            sleep(_RETRY_COOLDOWN)
+        raise RuntimeError("Unable to execute " + method.__name__ + " after " + str(_RETRY_LIMIT) + " retries.")
+
+    def get_git_time(self):
+        datetime_string = self._try_git(self.git.get_api_status).raw_headers['date']
+        try:
+            datetime_object =  datetime.datetime.strptime(datetime_string, '%a, %d %b %Y %H:%M:%S %Z')
+        except:
+            log.exception("Failed to parse date retrieved from GitHub: %s", str(datetime_string))
+            raise
+        return datetime_object
+
+    @property
```

Comment:
Why is this a property? `get_pull_requests` would be a much more communicative name for this function.

Reply:
`get_pull_requests` is more like name for Java function if we would decide to use `@property` i would stay with `pull_requests`

Reply:
You are creating an interface which relies on network connectivity, but hiding it in an API method which looks very much like a local property. This is going to work, but it's confusing API design.